Add query.sql file with tasks.

grigorii-ch opened the pull request.

xgrisha committed:
Add query.sql file with tasks.

tlt-java-courses commented on blank-project-db/src/main/resources/query.sql:
```diff
@@ -0,0 +1,19 @@
+
+select * from orders where orderdate between '2004-01-05' and '2004-02-05';
+
+select sum(1), customerid from orders group by customerid order by 1 desc;
```
Тут вместо функции sum правильней использовать count(*)

grigorii-ch replied:
поправил

tlt-java-courses commented on blank-project-db/src/main/resources/query.sql:
```diff
@@ -0,0 +1,19 @@
+
+select sum(1), customerid from orders group by customerid order by 1 desc;
+
+select sum(totalamount) from orders group by customerid, orderdate having sum(totalamount) > 100; 
```
тут нужно еще и в списке полей указать поля по которым группируешь

grigorii-ch replied:
добавил

tlt-java-courses commented on blank-project-db/src/main/resources/query.sql:
```diff
@@ -0,0 +1,19 @@
+
+select * from products where price between 10 and 20;
+
+select * from orders where orderdate between '2004-01-05' and '2004-02-05';
```
При работе с датыми, лучше приводить литерал к типу date с помощью функции to_date

grigorii-ch replied:
поправил

xgrisha committed:
refactoring
added:
 - to_date for date literals
 - change sun() to count()
 - added fields to query section
